feat(docs): add python and poetry plugin docs (#1940)
canonical/charmcraft: docs/reference/plugins/poetry_plugin.rst

.. _craft_parts_poetry_plugin:

Poetry plugin
=============

The Poetry plugin can be used for Python charms written using `Poetry`_ and the
`Operator framework`_.

.. include:: /common/craft-parts/reference/plugins/poetry_plugin.rst
   :start-after: .. _craft_parts_poetry_plugin-keywords:
   :end-before: .. _craft_parts_poetry_plugin-environment_variables:

python-keep-bins
~~~~~~~~~~~~~~~~
**Type**: boolean
**Default**: False

Whether to keep python scripts in the virtual environment's ``bin`` directory.

.. include:: /common/craft-parts/reference/plugins/poetry_plugin.rst
   :start-after: .. _craft_parts_poetry_plugin-environment_variables:
   :end-before: .. _poetry-details-end:

How it works
------------

During the build step, the plugin performs the following actions:

1. It creates a virtual environment in the
   :ref:`${CRAFT_PART_INSTALL}/venv <craft_parts_step_execution_environment>` directory.
2. It uses :command:`poetry export` to create a ``requirements.txt`` in the project's
   build directory.
3. It uses :command:`pip` to install the packages referenced in ``requirements.txt``

>    into the virtual environment, without any additional dependencies.

Reviewer: > additional dependencies.

I don't fully understand what "additional" suggests here. Can you elaborate?

Author: Dependencies not listed in requirements.txt.

Author: For example, if `requirements.txt` lists `pydantic` but not `pydantic-core`, it won't install `pydantic-core`.

> 3. It uses :command:`pip` to install the packages referenced in ``requirements.txt``
>    into the virtual environment, without any additional dependencies.

Reviewer: ```suggestion
3. It uses :command:`pip` to install the packages referenced in ``requirements.txt``
   into the virtual environment. Undeclared dependencies are ignored.
```

4. It copies the ``src`` and ``lib`` directories from your charm project into the

>    final charm if they exist.

Reviewer: This might flow better, and avoid implying `src && lib`, as:

```
4. It copies any existing ``src`` and ``lib`` directories from your charm project into the
   final charm.
```
